src/Feed/Feed.jsx

```diff
@@ -27,23 +45,88 @@ export default function Feed() {
         });
     }
 
-    fetchConsumerData().then(() =>
-      fs
-        .collection("recipe")
-        .get()
-        .then((querySnapshot) => {
+    const query = fs
+      .collection("recipe")
+      .where("allergies", "!=", selected.allergy);
+
+    // Clear recipes as firebase will filter recipes
+    setRecipes([]);
+
+    if (selected.mealtype === "") {
+      fetchConsumerData().then(() =>
+        query.get().then((querySnapshot) => {
           querySnapshot.forEach((doc) => {
             // Filter liked/disliked recipes
             if (!recipesToFilter.includes(doc.id)) {
               setRecipes((prevState) => [...prevState, doc.data()]);
             }
           });
         })
-    );
-  }, [fs, auth.currentUser.uid]);
+      );
+    } else {
+      fetchConsumerData().then(() =>
+        query
+          .where("mealType", "==", selected.mealtype)
+          .get()
+          .then((querySnapshot) => {
+            querySnapshot.forEach((doc) => {
+              // Filter liked/disliked recipes
+              if (!recipesToFilter.includes(doc.id)) {
+                setRecipes((prevState) => [...prevState, doc.data()]);
+              }
+            });
+          })
+      );
+    }
+  }, [fs, auth.currentUser.uid, selected]);
+
+  const onPreferencesSelect = (preferenceType, selectedPref) => {
+    setSelected((prevState) => ({
+      ...prevState,
+      [preferenceType]: selectedPref,
+    }));
+  };
+
+  const Preferences = allergies.map((allergy) => (
+    <Dropdown.Item
+      id="#preferences-dropdown-item"
+      key={allergy}
+      as="button"
+      onSelect={() => onPreferencesSelect("allergy", allergy)}
+    >
+      {allergy}
+    </Dropdown.Item>
+  ));
+
+  const Mealtypes = mealtypes.map((mealtype) => (
+    <Dropdown.Item
+      id="#preferences-dropdown-item"
+      key={mealtype}
+      as="button"
+      onSelect={() => onPreferencesSelect("mealtype", mealtype)}
+    >
+      {mealtype}
+    </Dropdown.Item>
+  ));
 
   return (
     <div id="feed">
+      <div id="preferences">
+        <DropdownButton
+          id="preferences-dropdown"
+          title="Allergies"
+          variant="danger"
```

Comment:
Danger doesn't sound nice. Might give implications semantically.

Reply:
This is the styling we use throughout the entire application, as it's the red colour. It's just called danger, but no need to get scared 😢